### PR TITLE
Imviz: Add a different radial profile approach

### LINKED ISSUE #1049: [FEAT] Imviz: add option for an integrated profile to the aperture photometry plugin
### Description

The current aperture photometry plugin shows pixel values.  This is sometimes exactly what the user wants, but other times they want the integrated flux profile (sometimes called the "curve of growth"). We should add an option to the aperture photometry plugin to do exactly that - i.e., it would toggle the plot between the current "pixel plot" behavior and an integrated profile.

The idea would be to basically have it plot exactly what the algorithm I describe in #1048 does, but actually plot the profile from steps 1-3 instead of using it to compute the FWHM.  That has the convenient feature that the user could look at that plot and immediately see what the FWHM is (i.e., it's where the profile first crosses "half of the max").  Could even add a thin horizontal line at the half-max, although that might make the plot too busy (I leave it to the implementors discretion for the first pass at least).

### Additional context

Brought up out-of-band in a discussion with @pontoppi, who may want to comment here if there's more to add than what I have above.

[🐱](https://jira.stsci.edu/browse/JDAT-2100)

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request is to address the need to add more plot options to the simple aperture photometry plugin in Imviz.

Partially fixes #1049 although a new ticket needs to be made to implement curve of growth. (The rest of #1049 will be part of #1048 anyway.)

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Since radial profile is actually not a released feature, we don't need a change log here.

Curve of growth is being discussed at astropy/photutils#1298 .

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py
Comment (pllim) [inline jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:269]: Is the casting to `list` necessary here?

**Review by pllim — COMMENTED** [review]
Thanks! I am definitely in favor of this PR since you have shown the performance boost by calculating the radial profile this way.

This code is from `imexam`, right? If so, you need to copy the [imexam license](https://github.com/spacetelescope/imexam/blob/master/LICENSE.rst) to `licenses/IMEXAM_LICENSE.rst` here. Also please indicate clearly in the code where appropriate that the algorithm is from imexam and don't forget "see licenses/IMEXAM_LICENSE.rst" in that comment.

File: jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py
Comment (javerbukh) [reply jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py:269]: Surprisingly yes! This line throws an error without it, something about a `typeerror` between 64 bit and 32 bit numbers.

**Comment by pllim** [comment]
I applied my review directly as a second commit. The plot now looks like this:

Screenshot 2022-02-22 150015

I don't think this needs a second approval since Jesse already presented this to POs/devs earlier this week and received positive feedback already. I will merge when CI passes. Thanks!

**Review by pllim — APPROVED** [review]

**Pull request merged**
